internal/digraph/scheduler/node.go

```diff
@@ -400,7 +400,14 @@ func (n *Node) Setup(ctx context.Context, logDir string, dagRunID string) error
 	safeName := fileutil.SafeName(n.Name())
 	timestamp := startedAt.Format("20060102.15:04:05.000")
 	postfix := stringutil.TruncString(dagRunID, 8)
-	logFilename := fmt.Sprintf("%s.%s.%s", safeName, timestamp, postfix)
+
+	// Add retry count to filename if this is a retry
+	retrySuffix := ""
+	if n.GetRetryCount() > 0 {
+		retrySuffix = fmt.Sprintf(".retry%d", n.GetRetryCount())
+	}
```

Comment:
Question: Why do we need to create separate log files? I'm asking because if we create different log files, the older ones might become orphaned and won't be removed when Dagu deletes the DAG-run data.